src/Workspaces/MSBuild/Core/MSBuild/PathResolver.cs

```diff
@@ -11,9 +11,9 @@ namespace Microsoft.CodeAnalysis.MSBuild;
 
 internal sealed class PathResolver
 {
-    private readonly DiagnosticReporter _diagnosticReporter;
+    private readonly DiagnosticReporter? _diagnosticReporter;
 
-    public PathResolver(DiagnosticReporter diagnosticReporter)
+    public PathResolver(DiagnosticReporter? diagnosticReporter)
```

Comment:
Made the `DiagnosticReporter` optional since this is also used by the `LanguageServerProjectSystem` which does not use a reporter.

Reply:
Should we be though?